### PR TITLE
Refactor authentication flow

### PR DESCRIPTION
**Pull request by ttaoS**

### CODE REVIEW

File: src/ts/routes/authenticated/authenticated.tsx [hunk 1]
```diff
@@ -3,20 +3,23 @@ import * as React from 'react'
...
   return {
     welcome: state.settings.welcome,
-    isLocked: state.account.locked
+    isLocked: state.account.locked,
+    accountCreated: state.account.created
   }
 }
 
 export class AuthenticatedRoute extends React.Component<StateProps & IRouteProps & RouteProps> {
   renderRoute (props: StateProps & IRouteProps & RouteProps) {
```
Comment (fyang1024) [inline src/ts/routes/authenticated/authenticated.tsx]: probably it's worth doing 
`const { isLocked, welcome, accountCreated } = props` here
and then remove `props.` prefixes below ?

File: src/ts/components/account/Term.tsx [hunk 2]
```diff
@@ -13,8 +13,9 @@ class Term extends React.Component<ITermProp> {
 
   handleClick = () => {
     const { history, settings } = this.props
-    this.props.saveSettings({ ...settings, welcome: false })
-    history.push(CREATE_PASSWORD_ROUTE)
+    this.props.saveSettings({ ...settings, welcome: false }).then(() =>
+      history.push(DEFAULT_ROUTE)
```
Comment (mytechtip) [inline src/ts/components/account/Term.tsx]: Does that mean now the `DEFAULT_ROUTE` will now check if it goes to login or create new password?

Comment (ttaoS) [reply src/ts/components/account/Term.tsx]: yeah i probably should rename it to home_route

File: src/ts/routes/authenticated/authenticated.tsx (hunk 1 above)
Comment (ttaoS) [reply src/ts/routes/authenticated/authenticated.tsx]: upated

**Pull request merged**